eclair-core/src/main/scala/fr/acinq/eclair/io/MessageRelay.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 ACINQ SAS
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.io
+
+import akka.actor.{Actor, ActorRef, ActorRefFactory, Props}
+import fr.acinq.bitcoin.Crypto.PublicKey
+import fr.acinq.eclair.wire.protocol.OnionMessage
+
+object MessageRelay {
+  case object Relay
+
+  def relay(context: ActorRefFactory, switchboard: ActorRef, nextNodeId: PublicKey, msg: OnionMessage): Unit = {
+    val actor = context.actorOf(Props(new MessageRelay(switchboard, nextNodeId, msg)))
+    actor ! Relay
+  }
+}
+
+class MessageRelay private(switchboard: ActorRef, nextNodeId: PublicKey, msg: OnionMessage) extends Actor {
+
+  import MessageRelay._
+
+  override def receive: Receive = {
+    case Relay =>
+      switchboard ! Peer.Connect(nextNodeId, None)
+    case PeerConnection.ConnectionResult.Connected | PeerConnection.ConnectionResult.AlreadyConnected =>
+      sender() ! msg
+      context.stop(self)
```

Comment:
I'm not sure yet, but I don't think we should stop ourselves immediately.
We should probably watch the peer actor and stop ourselves when they stop.
This way later we can implement a mechanism to have this `MessageRelay` actor actively stop the peer after a while if the `Peer` is inactive (not sure of the details of how we'll do that yet though).

Reply:
Done

Reply:
It's true that we don't need it yet, we can introduce that when we implement automatically stopping these ephemeral peers.